[nnc] Add LoopNest::getLoopAt to retrieve a specified inner For-stmt

huiguoo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60601 [nnc] Add LoopNest::unsafe_fuseLoops to let users apply fusion on stmts that may violate our correctness checks
* **#60569 [nnc] Add LoopNest::getLoopAt to retrieve a specified inner For-stmt**
* #57758 [NNC] Add tile transformation in loopnest (fixed #52785)

Differential Revision: [D29337767](https://our.internmc.facebook.com/intern/diff/D29337767)

huiguoo committed:
[nnc] Add LoopNest::getLoopAt to retrieve a specified inner For-stmt

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29337767).

navahgar reviewed (commented):
LGTM. Can you also add a test for this?

I just noticed that none of our `get*` APIs in `LoopNest` have tests. We should definitely add tests for them as well (not in this PR).

huiguoo committed:
Update on "[nnc] Add LoopNest::getLoopAt to retrieve a specified inner For-stmt"


Differential Revision: [D29337767](https://our.internmc.facebook.com/intern/diff/D29337767)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29337767).

navahgar commented on test/cpp/tensorexpr/test_loopnest.cpp:
```diff
@@ -661,6 +661,58 @@ TEST(LoopNest, ExprSplitWithMaskRepeatedNoMask) {
+  auto for_j = For::make(j, 0, 100, Block::make({store1, for_k1, for_k2}));
+  auto for_i = For::make(i, 0, 100, for_j);
+  LoopNest l(Block::make({for_i}), {B, C});
+  auto ret_k2 = l.getLoopAt(for_i, {0, 2});
```
Can you also add a check here to ensure that `ret_k2` is same as `for_k2`?

huiguoo committed:
Update on "[nnc] Add LoopNest::getLoopAt to retrieve a specified inner For-stmt"


Differential Revision: [D29337767](https://our.internmc.facebook.com/intern/diff/D29337767)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29337767).
